File: 35-search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int minResult = Integer.MAX_VALUE;
        int maxResult = Integer.MIN_VALUE;
        int low = 0; int high = nums.length-1;
        while (low < high) {
            int m = ((int)Math.floor((high+low)/ 2)) + 1;
            if (nums[m] == target) {
                return m;
            } else if (target > nums[m]) {
                low = m;
            } else {
                high = m - 1;
            }
        }
        if (nums.length > 0 && nums[low] >= target) {
            return low;
        } else {
            return low + 1;
        }
    }
}

File: 35-search-insert-position/search-insert-position.swift
class Solution {
    func searchInsert(_ nums: [Int], _ target: Int) -> Int {
        var lo = 0, hi = nums.count - 1

        while lo < hi {
            var mid: Int = (hi + lo) / 2

            if nums[mid] == target {
                return mid
            } else if nums[mid] > target {
                hi = mid
            } else {
                lo = mid + 1
            }
        }
        if lo < nums.count{
            if nums[lo] < target {
                return lo + 1
            }
        }
        return lo
    }
}